### DISCUSSION TITLE
add_mesh_clip_plane update scalar bar fail

### DISCUSSION THREAD

**Discussion by giannoko** [pyvista/pyvista#3486] [Q&A]
Hi,

I add a mesh clip plane widget and I plot the "CellQuality" scalar bar. At the first window the cell quality is plotted, but when I change the plane, all cells have the same color. Here are pic samples

At start
![pic_1](https://user-images.githubusercontent.com/92250377/196393924-84c98c22-dc0a-41ed-a45e-1e4514d7146a.PNG)

After change the plane
![pic_2](https://user-images.githubusercontent.com/92250377/196393993-59948d48-fe23-4187-9f2d-1726ea710a95.PNG)

Here is the code 

`actor = p.add_mesh_clip_plane(
    inner,
    origin=(0.,0.,0.),
    normal='z',
    color = "white",
    style="surface",
    scalars="CellQuality",
    # clim = [0,1],
    show_edges=True,
    tubing=True,
    origin_translation=False,
    outline_translation=False,
    edge_color="black",
    opacity=1.0,
    lighting=False,
    cmap="plasma",
    label='Tube mesh',
    show_scalar_bar=True,
    scalar_bar_args={'title': 'Cell quallity' , 'color' : "blue"},
    render=True,
    crinkle=True
    
)
`
What can I do? 

Thank you

**Comment by banesullivan** [comment]
That seems odd. Could you share the mesh file?

**Reply by giannoko** [reply]
Hi @banesullivan. I attached the file mesh. Is there any problem with the version of .msh exported from Fluent?

**Reply by banesullivan** [reply]
Thanks for providing! I'm just now getting back to this...

**Comment by giannoko** [comment]
Hi @banesullivan. It is an .msh file exported from Fluent. I attach it as a .txt file because I was not able to upload it as .msh file.

[pipe.txt](https://github.com/pyvista/pyvista/files/9819471/pipe.txt)

**Comment by banesullivan** [comment]
I'm able to reproduce this. The issue appears to be with the `add_mesh_clip_plane` method messing up the active scalars on the mesh.

I'm seeing that after the initial clip, the active scalars are changed to "cell_ids" but the colormapping stays from 0-1 while the "cell_ids" range is 0-5.294e+05, explaining why everything looks yellow after the clip.

I believe this is happening when `crinkle=True` in this bit of code:

https://github.com/pyvista/pyvista/blob/c7f0efa8f9390a7c41f6d387d31322de311b4f1f/pyvista/plotting/widgets.py#L235

If I set `crinkle=False` in `add_mesh_clip_plane` the problem goes away.

This is a regression and reminds of the following open issues:

- https://github.com/pyvista/pyvista/issues/1897
- https://github.com/pyvista/pyvista/issues/3293

For now, I recommend setting `crinkle=False`


@pyvista/developers, I want to highlight this issue as I think it is a part of https://github.com/pyvista/pyvista/issues/1897 and a reason why we should make sure that adding an array does not set the array as active by default

**Reply by giannoko** [reply]
Hi @banesullivan, Yes you are right. I set "crinkle=False" and now everything is ok. Thank you!

**Reply by banesullivan** [reply]
This should get resolved by #3535
